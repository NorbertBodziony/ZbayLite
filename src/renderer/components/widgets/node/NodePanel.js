import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import { withStyles } from '@material-ui/core/styles'

import ExpandLessIcon from '@material-ui/icons/ExpandLess'

import NodeStatus from '../../../containers/widgets/node/NodeStatus'
import NodePanelDetails from '../../../containers/widgets/node/NodePanelDetails'
import Tooltip from '../../ui/Tooltip'

const styles = theme => ({
  expansionDetails: {
    padding: 0,
    margin: 0
  },
  expander: {
    boxShadow: 'none',
    color: theme.palette.colors.white,
    backgroundColor: 'transparent',
    padding: 0,
    margin: 0,
    width: '100%'
  },
  root: {
    position: 'relative'
  },
  icon: {
    color: 'inherit',
    fontSize: 15,
    marginLeft: -10,
    marginRight: 0
  },
  panelSummary: {
    padding: 0,
    margin: 0,
    height: 35,
    minHeight: 15,
    color: 'rgba(255, 255, 255, 0.5)',
    '&:hover': {
      color: 'rgba(255, 255, 255, 1)'
    }
  },
  expanded: {
    backgroundColor: 'rgb(0,0,0,0.3)'
  },
  expandedSummary: {
    minHeight: '15px !important'
  },
  tooltip: {
    marginRight: 10,
    marginTop: 0
  },
  baseNotExpanded: {
    width: '80px'
  },
  baseExpanded: {
    width: '100%'
  }
})

export const NodePanel = ({ classes, expanded, setExpanded }) => (
  <Grid
    item
    container
    justify='flex-end'
    className={classNames({
      [classes.root]: true
    })}
  >
    <Grid
      item
      className={classNames({
        [classes.baseExpanded]: expanded,
        [classes.baseNotExpanded]: !expanded
      })}
    >
      <ExpansionPanel
        expanded={expanded}
        onClick={() => setExpanded(!expanded)}
        square
        classes={{ root: classes.expander, expanded: classes.expanded }}
      >
        <Tooltip title='Node status' className={classes.tooltip} placement='bottom-end'>
          <ExpansionPanelSummary
            classes={{
              root: classes.panelSummary,
              expandIcon: classes.icon,
              expanded: classes.expandedSummary
            }}
            expandIcon={<ExpandLessIcon fontSize='inherit' />}
          >
            <NodeStatus />
          </ExpansionPanelSummary>
        </Tooltip>

        <ExpansionPanelDetails className={classes.expansionDetails}>
          <NodePanelDetails expanded={expanded} />
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </Grid>
  </Grid>
)

NodePanel.propTypes = {
  classes: PropTypes.object.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(NodePanel)
