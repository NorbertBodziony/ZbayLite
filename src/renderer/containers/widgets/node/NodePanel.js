import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelComponent from '../../../components/widgets/node/NodePanel'
import nodeHandlers from '../../../store/handlers/node'
import { useInterval } from '../../hooks'

export const mapDispatchToProps = dispatch => bindActionCreators({
  getStatus: nodeHandlers.epics.getStatus
}, dispatch)

export const NodePanel = ({ className, getStatus }) => {
  useInterval(getStatus, 15000)
  return (
    <NodePanelComponent className={className} />
  )
}

export default connect(null, mapDispatchToProps)(NodePanel)
