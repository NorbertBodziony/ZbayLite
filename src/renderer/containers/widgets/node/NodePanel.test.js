/* eslint import/first: 0 */
jest.mock('../../hooks', () => ({ useInterval: jest.fn() }))

import React from 'react'
import { shallow } from 'enzyme'

import { useInterval } from '../../hooks'
import { mapDispatchToProps, NodePanel } from './NodePanel'

describe('NodePanel', () => {
  it('will receive right actions', async () => {
    const actions = mapDispatchToProps(x => x)
    expect(actions).toMatchSnapshot()
  })

  it('sets interval for getStatus', () => {
    shallow(<NodePanel getStatus={jest.fn()} />)
    expect(useInterval.mock.calls).toMatchSnapshot()
  })
})
