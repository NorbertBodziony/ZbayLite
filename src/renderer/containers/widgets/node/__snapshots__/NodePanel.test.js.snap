// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodePanel sets interval for getStatus 1`] = `
Array [
  Array [
    [MockFunction],
    15000,
  ],
]
`;

exports[`NodePanel will receive right actions 1`] = `
Object {
  "getStatus": [Function],
}
`;
